### DISCUSSION TITLE
How to reconnect a mobile as quickly as possible to socket.io?

### DISCUSSION THREAD

**Discussion by Madriix** [socketio/socket.io#4335] [Q&A]
Hi

On an Android mobile, I checked "Energy saving" in the options. I noticed that getting I disconnected very quickly from socket.io in 2 methods:

- either you disconnect immediately with the mobile screen off
- either we start another application or we go to another window of another native app, and a few seconds later we disconnect from our app (socket.io).

I'm looking for a way to reconnect as soon as possible to socket.io

I noticed that by putting a popup in 'disconnect' (equivalent to alert() ) I noticed that this popup is displayed as soon as I display my application, which means that the "on disconnect" is detected immediately, I'm looking for a way to automatically reconnect to the application as quickly as possible, if possible in 1 second, but I can't.
Instead it reconnects to socket.io about 10 even 30 seconds later.

Anyone know what's wrong with my app?

Here is my structure:

```
(function (global, undefined) {

  function Client() {
    if (!(this instanceof Client)) {
      return new Client();
    }
    //...
  }

  Client.prototype.connect = function (server) {
    if ($('#loader').length)
      $('#loader').html("Connexion en cours...");

    this.panels.get('status').write(null, '[NOTICE]', 'Connexion en cours...');

    this.socket = io(server, {
      closeOnBeforeunload: false,
      reconnection: true,
      reconnectionDelay: 1000,
      reconnectionAttempts: 10
    });

    this.socket.on("connect_error", () => {
      setTimeout(() => {
        this.socket.connect();
      }, 1000);
    });

    this.socket.on('connect', function () {
      if (this.socket.connected == true)
        this.panels.get('status').write(null, '[NOTICE]', 'Socket connected');
      else
        this.panels.get('status').write(null, '[NOTICE]', 'Socket false');

      var $this = this;

      this.socket.emit('app_connect', {
        nick: (ClientSettings.nick || null),
      }, function (data) {});


      if (global.connected == false) {
        global.connected = true;
      }

    }.bind(this));

    this.socket.on('disconnect', function (reason) {

      // my popup is here

      this.socket.disconnect().connect();

    }.bind(this));




    $('body').on('click', '.report_abuse', (function (parent) {
      return function () {
        //...
      }
    })(this));


    // ...


    return this;
  };

  global.onload = function () {
    //...
    if (!global['Client']) {
      (global['Client'] = Client()).connect(ClientSettings.socketio);
      //...
    }
  };

})(window);
```

Cdt

**Comment by darrachequesne** [comment]
Hi! I think you can either:

- reduce the value of the `reconnectionDelayMax` option, which defaults to 5000

This is the upper bound for the exponential reconnection delay.

Reference: https://socket.io/docs/v4/client-options/#reconnectiondelaymax

- listen to the `focus` window event

```js
window.addEventListener('focus', () => {
  socket.connect();
});
```

Reference: https://developer.mozilla.org/en-US/docs/Web/API/Window/focus_event

**Comment by Madriix** [comment]
Thank, I set reconnectionDelayMax to 3000. 

I just noticed on my iPhone (Safari) that it reconnected quickly (between 1 and 3 seconds) but on my Samsung Android (Chrome) it was a little longer, sometimes 5 or 10 or 15 seconds. 

Maybe JavaScript is still frozen (stopped) on Android as soon as the phone boots up, the login call is done before JavaScript is unblocked/unfreezed from the browser or something like that.
